aviatrix/resource_aviatrix_azure_vng_conn.go

```diff
@@ -0,0 +1,110 @@
+package aviatrix
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
+)
+
+func resourceAviatrixAzureVngConn() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAviatrixAzureVngConnCreate,
+		Read:   resourceAviatrixAzureVngConnRead,
+		Delete: resourceAviatrixAzureVngConnDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"primary_gateway_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Primary gateway name",
+			},
+			"connection_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Connection name",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "VPC ID",
+			},
+			"vng_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "VNG name",
+			},
+			"attached": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "VNG attached or not",
+			},
+		},
+	}
+}
+
+func marshalAzureVngConnInput(d *schema.ResourceData) *goaviatrix.AzureVngConn {
+	return &goaviatrix.AzureVngConn{
+		PrimaryGatewayName: d.Get("primary_gateway_name").(string),
+		ConnectionName:     d.Get("connection_name").(string),
+	}
+}
+
+func resourceAviatrixAzureVngConnCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	azureVngConn := marshalAzureVngConnInput(d)
+
+	if err := client.ConnectAzureVng(azureVngConn); err != nil {
+		return fmt.Errorf("could not connect to azure vng: %v", err)
+	}
+
+	d.SetId(azureVngConn.ConnectionName)
+	return resourceAviatrixAzureVngConnRead(d, meta)
+}
+
+func resourceAviatrixAzureVngConnRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	connectionName := d.Get("connection_name").(string)
+
+	if connectionName == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import. Import Id is %s", id)
+
+		d.Set("connection_name", id)
+		connectionName = id
+	}
+
+	azureVngConnStatus, err := client.GetAzureVngConnStatus(connectionName)
+	if err != nil {
+		return fmt.Errorf("could not get azure vng conn status: %v", err)
+	}
```

Comment:
From HashiCorp Terraform Provider Tutorial (https://learn.hashicorp.com/tutorials/terraform/provider-setup):

> When you create something in Terraform but delete it manually, Terraform should gracefully handle it. If the API returns an error when the resource doesn't exist, the read function should check to see if the resource is available first. If the resource isn't available, the function should set the ID to an empty string so Terraform "destroys" the resource in state. The following code snippet is an example of how this can be implemented; you do not need to add this to your configuration for this tutorial.
```
if resourceDoesntExist {
  d.SetID("")
  return
}
```

In our provider we use a specific error to signify that there was no actual error, just the resource was not found `goaviatrix.ErrNotFound`. For example in `dataSourceAviatrixFirewallRead`
```
	fw, err := client.GetPolicy(firewall)
	if err == goaviatrix.ErrNotFound {
		d.SetId("")
		return nil
	}
	if err != nil {
		return fmt.Errorf("error fetching firewall policy for gateway %s: %s", firewall.GwName, err)
	}
```


If you haven't already, please take the time to go through the entire Terraform provider tutorial.